vignettes_src/troubleshooting.Rmd

```diff
@@ -242,3 +242,56 @@ obj$reconcile()
 ```
 
 Which will print information about anything that was adjusted.
+
```

Comment:
Some ideas:

# My job works on my computer but not on the cluster.

In that case, something is different between how the cluster sees the world, and how your computer sees it. 

* Look in the logs to try and find the reason why the failing jobs are doing so.
* Are there variables in the global R environment on your local computer, that your code relies upon, that won't be present on the cluster? Do you have local R packages or sources loaded which you haven't declared when initialising your context?
* Or any system variables, or other dependencies which enable your job to work locally, but won't be set up on a cluster node?
* Are you referring to any files visible to your local machine, but not on the cluster? Are you referring to `C:` for instance?
* (Rarely:) Are you viewing a cached version of any network storage on your local computer, that has not been synced to the real network storage view that the cluster has?
* Check that you have not run out of disk-space. The Q: quota is normally 15Gb.
* If you are running C code, check for other causes of indeterminate failures, such as uninitialised variables, or array out-of-bounds errors. These are unpredictable errors by nature, but surprisingly often you might get away with it on a local computer, while a cluster node behaves differently.
* If you are running stochastic code, check that you are *really* using the same random number seeds.


# Some of my jobs work on the cluster, but others fail. 

* Look in the logs to try and find the reason why the failing jobs do so. 
* Try rerunning the failed jobs. Is it the same set that passes, and the same set that fails? In that case, consider what makes your jobs different - pehaps job-specific input data or parameters could cause the failures.
* If you find messages about "Error allocating a vector..." or "std::bad_alloc", then try and work out the memory usage of a single job. Perhaps run it locally with task manager (windows), or top (linux) running, and watch to see what the memory usage is. If the job is single-core, consider the total memory used if you run 8 or 16 instances on the same cluster machine. If the total memory exceeds the available, then behaviour will be undefined, and some jobs will likely fail.
* In the above example, note that someone else's memory-hungry combination of jobs may affect your small-memory job, if they run on the same node. We don't enforce any memory limits on jobs, which on the whole, is nice and convenient, but it carries the risk that the above can happen.
* Always check you're not running out disk space. The Q: quota is normally 15Gb.
* Find what node your jobs were running on. If you consistently get errors on one node, but not others, then get in touch with Wes, as we do get node failures from time to time, where the fault is not obvious at first.

# My job is slower on the cluster than running locally!

* This is expected, especially for single-core jobs. Cluster nodes are aiming to provide bandwidth, rather than linear performance, so a single job may run slower on a cluster node than on your own computer. But the cluster node might be able to run 8 or more such jobs at once, without taking any longer, while you continue using your local computer for local things. 
* If that is still insufficient, and you still want to compare timings in this way, then check that the cluster is doing *exactly* the same work as your local computer.